packages/interfaces/src/libp2p.ts

```diff
@@ -0,0 +1,21 @@
+import type { GossipSub } from "@chainsafe/libp2p-gossipsub";
+import type { Libp2p as BaseLibp2p } from "@libp2p/interface-libp2p";
+import type { Libp2pInit } from "libp2p";
+import type { identifyService } from "libp2p/identify";
+import type { PingService } from "libp2p/ping";
+
+export type Libp2pServices = {
+  ping: PingService;
+  pubsub?: GossipSub;
+  identify: ReturnType<ReturnType<typeof identifyService>>;
+};
+
+// TODO: Get libp2p to export this.
+export type Libp2pComponents = Parameters<
```

Comment:
This is a trick to have `Libp2pComponents` type on our side. I will follow up with some issue for them to allow export of it.

Reply:
Did you already open an issue on their side?

Reply:
Created issues [here](https://github.com/libp2p/js-libp2p/issues/1875)